pkg/autoscaler/metrics/wire.proto

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto2";
+package metrics;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+option (gogoproto.goproto_getters_all) = false;
+
+message WireKey {
+    required string namespace = 1;
+    required string name = 2;
+}
+
+message WireStat {
+  required string podName = 2;
+
+  required double processUptime = 3;
+  required double averageConcurrentRequests = 4;
+  required double averageProxiedConcurrentRequests = 5;
+  required double requestCount = 6;
+  required double proxiedRequestCount = 7;
```

Comment:
do we definitely want these all to be required? Asking for two reasons (1) a lot of protobuf people advise against required fields I think (e.g. https://developers.google.com/protocol-buffers/docs/style) bc once it's required you can't un-require it easily (2) It seems at least concievable that we might decide to optimise by e.g. not bothering splitting out proxied requests / not sending whichever of the metrics isn't being used / not sending podName for direct scraped requests etc

Reply:
Fair enough! I looked at this mostly from an efficiency perspective and non-pointers are great for less allocations. I agree though that it might be worth spending the extra nanos in favor of having a better contract. I'll have another look.

Reply:
yeah, you may well be right that the trade-off of the extra GC isn't worth it - it looks though that gogoproto [might be smart enough](https://godoc.org/github.com/gogo/protobuf/gogoproto) to avoid generating pointers even for optional fields if `[(gogoproto.nullable) = false]` is set in which case it _may_ be possible to take the Brexit approach to this cake ;).

Reply:
Switched over to proto3 since so this is obsolete I think.